### ISSUE TITLE
AMS wont save changes on Filament type

### ISSUE THREAD

**Issue by Quameda** [bambulab/BambuStudio#928]
If you change the AMS slots to self generated Material and restart Bambulab studio the changes wil not be saved. At restart the slots are all Generic.
V 1.4.0.18 OSX

**Comment by bambu123** [comment]
What is self-generated Material? Currently, only the system filament preset can be set to the AMS tray. Also, you can't edit the information on the Bambu Office filament.

**Comment by Quameda** [comment]
Hi,

wasn’t able to send images would not let me upload. So here they are.

Setting these in the studio:


￼


then after restart of the studio it looks like this so it is NOT saved:

￼


So I have to klick and chose 8 at the moment and soon 16 !!!! Colors each and every time or I load a dummy project where the filament is saved. On the Pribter I can not choose and set my eSUN presets

This is not nice and not acceptable to be honest.

kind regards,

Peter Balogh

**Comment by bambu123** [comment]
Can't see the pictures, are you mean the filament-type setting here?
![image](https://user-images.githubusercontent.com/112537880/210743710-0cf31bb9-1cf5-4eb2-85a2-e574d8acf634.png)

**Comment by Quameda** [comment]
yes if you set a manually edited and saved filament type and you reopen the Studio the setting is gone. Should not be hard to save those.

kind regards,

Peter

**Comment by bambu123** [comment]
But manually edited and saved filament type (it's really preset) can't go to the Combobox here.
The list only has system presets.
![image](https://user-images.githubusercontent.com/112537880/210747873-3aaefab7-5d7e-4c4c-bb3d-0297776edaa5.png)

If you set 'Bambu PLA Basic' here, is it saved?

**Comment by Quameda** [comment]
only Bambupresets are saved. If they are edited they are not which is really annoying

**Comment by bambu123** [comment]
The modification to the system preset can not be saved to the AMS tray. It is only the preset id been saved to the AMS tray.
We plan to support user preset here, but currently, only system preset can be selected.

**Comment by Quameda** [comment]
if you can save it in a project, dummy project is workaround right now, you can save it in Studio, no?

**Comment by bambu123** [comment]
We can save the list of filament with the preset (type) and color to project. But it is not related to AMS. So do you mean the filament list is not saved by Studio? [image]

**Comment by Quameda** [comment]
why cant you see images in the email? Would be easier to explain.

Am 05.01.2023 um 12:07 schrieb cmguo ***@***.***>:

﻿

We can save the list of filament with the preset (type) and color to project. But it is not related to AMS. So do you mean the filament list is not saved by Studio? [image]

**Comment by bambu123** [comment]
You should attach image from github web page other than from email.

Add comment in this link.
https://github.com/bambulab/BambuStudio/issues/928

**Comment by bambu123** [comment]
@Quameda Please help continue to provide information, especially screenshots of the program.

**Comment by Quameda** [comment]
This is when you set and leave Bambustudio
![1](https://user-images.githubusercontent.com/50173018/212241178-75ffe87b-05e3-45cf-a747-013b905940f9.jpg)

This is when you start it again

![image](https://user-images.githubusercontent.com/50173018/212241386-e1a3d6f5-7191-4f34-814e-6063fbd6a573.png)

**Comment by bambu123** [comment]
I know this is a known issue with custom user presets that are not loaded at the very beginning of the program startup.
We will release a new version soon which fix this issue.

**Comment by Quameda** [comment]
Ok great will you also synchronize it back to the AMS? Would be nice to set everything on PC and have one setting in the Studio and when you stand at the Printer.

**Comment by bambu123** [comment]
Sorry, it is not related to AMS. To modify AMS filament info, please go to the device panel.  Also. custom filament preset can't be written to the AMS tray until the last version.

**Comment by Quameda** [comment]
That’s the problem AMS and Studio Settings are related of course and should have the same settings and communicate both ways. Why should there be another setting on the AMS than in studio?
